airbyte-integrations/connectors/destination-mssql/src/main/resources/spec.json

```diff
@@ -0,0 +1,90 @@
+  "supportsIncremental": true,
+  "supported_destination_sync_modes": ["overwrite", "append"],
+  "connectionSpecification": {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Postgres Destination Spec",
+    "type": "object",
+    "required": ["host", "port", "username", "database", "schema"],
+    "additionalProperties": false,
+    "properties": {
+      "host": {
+        "title": "Host",
+        "description": "Hostname of the database.",
+        "type": "string",
+        "order": 0
+      },
+      "port": {
+        "title": "Port",
+        "description": "Port of the database.",
+        "type": "integer",
+        "minimum": 0,
+        "maximum": 65536,
+        "default": 5432,
+        "examples": ["5432"],
+        "order": 1
+      },
+      "database": {
+        "title": "DB Name",
+        "description": "Name of the database.",
+        "type": "string",
+        "order": 2
+      },
+      "schema": {
+        "title": "Default Schema",
+        "description": "The default schema tables are written to if the source does not specify a namespace. The usual value for this field is \"public\".",
+        "type": "string",
+        "examples": ["public"],
+        "default": "public",
+        "order": 3
+      },
+      "username": {
+        "title": "User",
+        "description": "Username to use to access the database.",
+        "type": "string",
+        "order": 4
+      },
+      "password": {
+        "title": "Password",
+        "description": "Password associated with the username.",
+        "type": "string",
+        "airbyte_secret": true,
+        "order": 5
+      },
+      "ssl": {
+        "title": "SSL Connection",
+        "description": "Encrypt data using SSL.",
+        "type": "boolean",
+        "default": false,
+        "order": 6
+      },
+      "trustServerCertificate": {
+        "title": "Trust Server Certificate",
+        "type": "boolean",
+        "default": false,
+        "description": "Whether or not to implicitly trust the server certificate without verification.  (Useful for local test databases.)",
+        "examples": [true, false],
+        "order": 7
+      },
+      "trustStoreName": {
+        "title": "Trust Store Name",
+        "type": "string",
+        "description": "The full path (including filename) to the certificate trustStore file, which contains the list of certificates that the client trusts.",
+        "order": 8
+      },
+      "trustStorePassword": {
+        "title": "Trust Store Password",
+        "type": "string",
+        "description": "The password used to check the integrity of the trustStore data.",
+        "order": 9
+      },
+      "hostNameInCertificate": {
+        "title": "Host Name In Certificate",
+        "type": "string",
+        "description": "Specifies the host name of the server. The value of this property must match the subject property of the certificate.",
+        "order": 10
+      }
+    }
+  }
```

Comment:
This seems like it should be a `oneOf` for clarity. I would suggest replacing this configuration with something that looks like this:

```
      "ssl_method": {
        "title": "SSL Method",
        "type": "object",
        "description": "Encryption method to use when communicating with the database",
        "order": 6,
        "oneOf": [
          {
            "title": "Unencrypted",
            "additionalProperties": false,
            "description": "Data transfer will not be encrypted.",
            "required": ["ssl_method"],
            "properties": {
              "ssl_method": {
                "type": "string",
                "enum": ["unencrypted"],
                "default": "unencrypted"
              }
            }
          },
          {
            "title": "Encrypted (trust server certificate)",
            "additionalProperties": false,
            "description": "Use the cert provided by the server.",
            "required": ["ssl_method"],
            "properties": {
              "ssl_method": {
                "type": "string",
                "enum": ["encrypted_trust_server_certificate"],
                "default": "encrypted_trust_server_certificate"
              }
            }
          },
          {
            "title": "Encrypted (verify certificate)",
            "additionalProperties": false,
            "description": "Data transfer will not be encrypted.",
            "required": ["ssl_method", "trustStoreName", "trustStorePassword"],
            "properties": {
              "ssl_method": {
                "type": "string",
                "enum": ["encrypted_verify_certificate"],
                "default": "encrypted_verify_certificate"
              },
              "trustStoreName": {
                "title": "Trust Store Name",
                "type": "string",
                "description": "The full path (including filename) to the certificate trustStore file, which contains the list of certificates that the client trusts.",
                "order": 8
              },
              "trustStorePassword": {
                "title": "Trust Store Password",
                "type": "string",
                "description": "The password used to check the integrity of the trustStore data.",
                "order": 9
              },
              "hostNameInCertificate": {
                "title": "Host Name In Certificate",
                "type": "string",
                "description": "Specifies the host name of the server. The value of this property must match the subject property of the certificate.",
                "order": 10
              }
            }
          }
        ]
      }
```

This displays in the UI as follows:

![Screen Shot 2021-05-12 at 3 15 19 PM](https://user-images.githubusercontent.com/9092207/118050850-e7967280-b334-11eb-8104-1941243a5afb.png)
![Screen Shot 2021-05-12 at 3 15 13 PM](https://user-images.githubusercontent.com/9092207/118050852-e8c79f80-b334-11eb-8674-bd783f2e78ba.png)
![Screen Shot 2021-05-12 at 3 15 08 PM](https://user-images.githubusercontent.com/9092207/118050853-e8c79f80-b334-11eb-98cf-9f6c46029a71.png)

What do you think? If you adopt this configuration, you'll have to adjust how the config is parsed in the `MSSQLDestination` and the config that gets passed in the test suites.